gateway-sp-comms/src/single_sp/update.rs

```diff
@@ -257,57 +258,142 @@ fn read_auxi_check_from_tlvc(data: &[u8]) -> Result<[u8; 32], UpdateError> {
     })
 }
 
+/// Isolate extraction of bootleby from old-format archives.
+// TODO: When old-format archives are eliminated from customer
+// racks and spares inventory, then this code can be removed.
+fn bootleby_from_old_style_archive(
+    image: Vec<u8>,
+    log: &Logger,
+) -> Result<Vec<u8>, UpdateError> {
+    // Try the pre-v1.2.0 Bootleby archive format.
+    let cursor = Cursor::new(image.as_slice());
+    let mut archive = zip::ZipArchive::new(cursor).map_err(|zip_error| {
+        // Return the original Hubris Archive error instead
+        // of our attempted zip extraction error.
+        HubtoolsError::ZipError(zip_error)
+    })?;
+
+    for i in 0..archive.len() {
+        match archive.by_index(i) {
+            Ok(mut file) => {
+                if file.name() == "bootleby.bin" {
+                    let mut rot_image = vec![];
+                    match file.read_to_end(&mut rot_image) {
+                        Ok(_) => {
+                            debug!(
+                                log,
+                                "using bootleby.bin from old-style archive"
+                            );
+                            return Ok(rot_image);
+                        }
+                        Err(err) => {
+                            error!(log, "cannot access bootleby.bin from zip file index {i}: {err}");
+                            break;
```

Comment:
Nit - we should probably return an error here; breaking means we'll return `UpdateError::ImageNotFound`, but that isn't the same as "there was a bootleby.bin but we failed to read it".

Reply:
I changed this to return UpdateError::InvalidArchive